examples/chip-tool/commands/common/Command.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ *   Copyright (c) 2020 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "Command.h"
+#include <sstream>
+#include <support/CHIPLogging.h>
+
+bool Command::InitArguments(int argc, char * argv[])
+{
+    bool isValidCommand = false;
+
+    SuccessOrExit(strcmp(name, argv[0]) != 0);
```

Comment:
This is making assumptions about the numeric value of `CHIP_NO_ERROR` which don't seem right to me.

Or put another way, it seems to me that we shouldn't be using `SuccessOrExit` on random booleans.  That's what `VerifyOrExit` is for, though in this case it would need an extra no-op arg....

Reply:
I removed this check in the new version of the code and I replaced and other usage that was making assumptions as you stated.